brainiak/utils/fmrisim.py

```diff
@@ -1006,18 +1387,163 @@ def mask_brain(volume,
+    # print('System noise:\t\t\t' + tmp)
+
+    return noise_dict
+
+
+def generate_noise(dimensions,
+                   stimfunction,
+                   tr_duration,
+                   mask=None,
+                   noise_dict={'overall': 0.1},
+                   ):
+    """ Generate the noise to be added to the signal.
+    Default noise parameters will create a noise volume with a standard
+    deviation of 0.1 (where the signal defaults to a value of 1). This has
+    built into estimates of how different types of noise mix. All noise
+    values can be set by the user
+
+    Parameters
+    ----------
+    dimensions : n length array, int
+        What is the shape of the volume to be generated
+
+    stimfunction :  Iterable, bool
+        When do the stimuli events occur
+
+    tr_duration : float
+        What is the duration, in seconds, of each TR?
+
+    mask : 4d array, float
+        The mask of the brain volume, using
+
+    noise_dict : dictionary, float
+        This is a dictionary that must contain the key: overall. If there
+        are no other variables provided then it will default values
+
+    Returns
+    ----------
+
+    multidimensional array, float
+        Generates the noise volume for these parameters
+
+    """
+
+    # Take in the noise dictionary and determine whether
+    noise_dict = _noise_dict_update(noise_dict)
+
+    # What are the dimensions of the volume, including time
+    dimensions_tr = (dimensions[0],
+                     dimensions[1],
+                     dimensions[2],
+                     int(len(stimfunction) / (tr_duration * 1000)))
+
+    # Get the mask of the brain and set it to be 3d
+    if mask is None:
+        mask = np.ones(dimensions_tr)
+
+    # Generate the noise
+    noise_temporal = _generate_noise_temporal(stimfunction=stimfunction,
+                                              tr_duration=tr_duration,
+                                              dimensions=dimensions,
+                                              mask=mask[:, :, :, 0],
+                                              spatial_sigma=noise_dict[
+                                                 'spatial_sigma'],
+                                              motion_sigma=noise_dict[
+                                                 'motion_sigma'],
+                                              drift_sigma=noise_dict[
+                                                 'drift_sigma'],
+                                              auto_reg_sigma=noise_dict[
+                                                 'auto_reg_sigma'],
+                                              physiological_sigma=noise_dict[
+                                                 'physiological_sigma'],
+                                              )
+
+    noise_system = _generate_noise_system(dimensions_tr=dimensions_tr,
+                                          ) * noise_dict['system_sigma']
```

Comment:
This style is appropriate for code that is expected to change. Do you expect to keep adding parameters to the `_generate_noise_system` call? Otherwise, consider this for readability:
```
noise_system = _generate_noise_system(dimensions_tr=dimensions_tr)
               * noise_dict['system_sigma']
```